articles/flask-crud-api/index.md

```diff
@@ -0,0 +1,306 @@
+```bash
+(venv)$ pip install pymysql
+```
+Lastly, we need to install flask-marshmallow and marshmallow libraries which eases the work of converting Python objects to JSON and JSON to Python objects. To install flask-marshmallow and marshmallow run the below command.
+```bash
+(venv)$ pip install flask-marshmallow marshmallow
+(venv)$ pip install marshmallow_sqlalchemy
+```
+
+Open the project folder in your favorite IDE and create a new file name `app.py` in the project folder.
+
+Paste the below code into the `app.py` file created above.
+```python
+from flask import Flask, request, jsonify, make_response
+from flask_sqlalchemy import SQLAlchemy
+from marshmallow import fields
+from marshmallow_sqlalchemy import ModelSchema
+```
+
+In the above section, we import all of the required modules that we will use in our application.
+`from flask import Flask` imports the Flask web module that we use to create an instance of the Flask web application.
+`from flask_sqlalchemy import SQLAlchemy` imports SQLAlchemy module which we use to connect to the database.
+`from marshmallow import fields` and `from marshmallow_sqlalchemy import ModelSchema` imports the modules that we will use to serialize Python objects.
+
+### Project configuration
+In this section, we create an instance of the flask application, setup the database configurations and create a database object.
+```python
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'mysql+pymysql://root:@localhost:3306/todo'
+db = SQLAlchemy(app)
+```
+`app = Flask(__name__)` creates an instance of the flask web application.
+
+`app.config['SQLALCHEMY_DATABASE_URI'] = 'mysql+pymysql://<username>:<password>@localhost:
+<database port>/<database name>'
+` contains the information required to connect to our database. Our database username is `root` ,the password is ` ` ,the database port is `3306` and the database name is `todo`.
+
+On your computer start the XAMPP control panel, start the apache and MySQL servers then visit `http://localhost/phpmyadmin/index.php` in your browser and create a database with the name `todo`.
+
+`db = SQLAlchemy(app)` creates an object of the SQLAlchemy and stores it in a variable `db`.
+
+### Creating the Todo model
+We are going to the `Todo` model which will be a Python class representing the `Todo` table in the database.
+
+```python
+# Model
+class Todo(db.Model):
+    __tablename__ = "todos"
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(20))
+    todo_description = db.Column(db.String(100))
+
+    def create(self):
+        db.session.add(self)
+        db.session.commit()
+        return self
+
+    def __init__(self, title, todo_description):
+        self.title = title
+        self.todo_description = todo_description
+
+    def __repr__(self):
+        return f"{self.id}"
+
+
+db.create_all()
+```
+`Todo` model class has an id property which is used as the primary key.</br> 
```

Comment:
Remove all the HTML breaks here. Instead, use the Markdown format of the breaks (skip one line)